### PR TITLE
Run bigquery queries in cleanup only for project's platforms.

### PR DESCRIPTION
**Pull request by inferno-chromium**

### CODE REVIEW

**Comment by inferno-chromium** [comment]
Causing 3x more queries for google instance [3 platforms] for every project. Should shorten cron.

File: src/appengine/handlers/cron/cleanup.py [hunk 1]
```diff
@@ -178,12 +181,11 @@ def cleanup_unused_fuzz_targets_and_jobs():
   ndb_utils.delete_multi(to_delete)
 
 
-def get_jobs_and_platforms_for_top_crashes():
-  """Return list of jobs and platforms to use for picking top crashes."""
-  jobs = set()
-  platforms = set()
-
+def get_projects_to_jobs_and_platforms_map_for_top_crashes():
```
Comment (oliverchang) [inline src/appengine/handlers/cron/cleanup.py]: nit: this is a little long and difficult to comprehend.

Maybe rename to `get_jobs_and_platforms_for_project`. The "top_crashes" part isn't really necessary context that needs to be included here in the name.

File: src/appengine/handlers/cron/cleanup.py [hunk 2]
```diff
@@ -178,12 +181,11 @@ def cleanup_unused_fuzz_targets_and_jobs():
...
+  """Return a map of projects to jobs and platforms map to use for picking top
+  crashes."""
   all_jobs = ndb_utils.get_all_from_model(data_types.Job)
+  projects_to_jobs_and_platforms_map = {}
```
Comment (oliverchang) [inline src/appengine/handlers/cron/cleanup.py]: nit: omit '_map' for brevity?

File: src/appengine/handlers/cron/cleanup.py (hunk 1 above)
Comment (inferno-chromium) [reply src/appengine/handlers/cron/cleanup.py]: Done!

File: src/appengine/handlers/cron/cleanup.py (hunk 2 above)
Comment (inferno-chromium) [reply src/appengine/handlers/cron/cleanup.py]: Done!

**Pull request merged**